service/session.cc

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2023-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include "service/session.hh"
+#include "log.hh"
+
+namespace service {
+
+static logging::logger slogger("session");
+
+static thread_local session_manager the_session_manager;
+
+session_manager& get_session_manager() {
+    return the_session_manager;
+}
+
+session_manager::session_manager() {
+    create_session(default_session_id);
```

Comment:
Why do we need the default session? Is there alternatives to get rid of it?

Reply:
It's to avoid special-casing for unguarded streaming. If we have default session, you can pass this session as the guard and materializing this guard will never fail because session is always there.

Reply:
Perhaps we can call it unguarded_session_id instead of default_session_id. And we can have a service::get_unguarded_session_id() to get it. It is clearer than "default" session.

Reply:
There are two levels of abstraction here, sessions and guards. The "unguarded" belongs to the guard. Default session is just a session which is always there. For the guard which is using it, we have `null_topology_guard`.